Orange/widgets/gui.py

```diff
@@ -2391,25 +2394,31 @@ def action(self, value):
 class CallFrontComboBox(ControlledCallFront):
     def __init__(self, control, valType=None, emptyString=""):
         super().__init__(control)
-        self.valType = valType
         self.emptyString = emptyString
 
     def action(self, value):
-        if value in ('', None):
-            value = self.emptyString
-        if self.valType:
-            for i in range(self.control.count()):
-                if self.valType(self.control.itemText(i)) == value:
-                    self.control.setCurrentIndex(i)
-                    return
-            if value:
-                log.warning("Unable to set %s to '%s'. Possible values are: %s",
-                            self.control, value,
-                            ', '.join(self.control.itemText(i)
-                                      for i in range(self.control.count())))
+        def action_str():
+            items = [combo.itemText(i) for i in range(combo.count())]
+            try:
+                index = items.index(value or self.emptyString)
+            except ValueError:
+                log.warning("Unable to set '{}' to '{}'; valid values are '{}'".
+                            format(self.control, value, ", ".join(items)))
+            else:
+                self.control.setCurrentIndex(index)
+
+        def action_int():
+            if value < combo.count():
+                combo.setCurrentIndex(value)
+            else:
+                log.warning("Unable to set '{}' to {}; largest index is {}".
+                            format(combo, value, combo.count() - 1))
+
+        combo = self.control
+        if isinstance(value, int):
```

Comment:
This allows using both, integers and strings. I'll probably split this class in two in further commits; if `valueType` is `str`, the widget should not set the corresponding attribute to an `int`. The current implementation, however, does not hurt, it just allows broken widgets to work.